src/app/shared/components/header/header.component.ts

```diff
@@ -1,29 +1,22 @@
-import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+import { Component, OnInit } from '@angular/core';
+import { Router, RoutesRecognized } from '@angular/router';
 
 @Component({
   selector: 'app-header',
   templateUrl: './header.component.html',
   styleUrls: ['./header.component.scss'],
 })
 export class HeaderComponent implements OnInit {
-  @Input()
-  minHeader: boolean = true;
+  public minHeader: boolean = false;
 
-  @Output()
-  loginClickEvent = new EventEmitter();
+  constructor(private router: Router) {}
 
-  @Output()
-  logoutClickEvent = new EventEmitter();
-
-  constructor() {}
-
-  ngOnInit(): void {}
-
-  onLoginClick(): void {
-    this.loginClickEvent.emit();
-  }
-
-  onLogoutClick(): void {
-    this.logoutClickEvent.emit();
+  ngOnInit(): void {
+    this.router.events.subscribe((data) => {
```

Comment:
Please, follow the comment above unsubscription from observables. And there is another [good article](https://netbasal.com/when-to-unsubscribe-in-angular-d61c6b21bad3) for you ;)

Reply:
unsubscription added